# Рынок видеоигр

- Автор: Захаров Максим
- Дата: 21.12.2025

### Цели и задачи проекта

<font color='#777778'>В этой тетрадке используется датасет `/datasets/new_games.csv.`,который содержит информацию о продажах игр разных жанров и платформ,а также пользовательские и экспретные оценки игр.На примере датасета мы познакомимся с данными,проверим их корректность и проведем предобработку,получив необходимый срез данных. </font>

### Описание данных

<font color='#777778'>В проекте будут использованы данные датасета `/datasets/new_games.csv.`с таким описанием:
    
- Name — название игры.
- Platform — название платформы.
- Year of Release — год выпуска игры.
- Genre — жанр игры.
- NA sales — продажи в Северной Америке (в миллионах проданных копий).
- EU sales — продажи в Европе (в миллионах проданных копий).
- JP sales — продажи в Японии (в миллионах проданных копий).
- Other sales — продажи в других странах (в миллионах проданных копий).
- Critic Score — оценка критиков (от 0 до 100).
- User Score — оценка пользователей (от 0 до 10).
- Rating — рейтинг организации ESRB (англ. Entertainment Software Rating Board). Эта ассоциация определяет рейтинг компьютерных игр и присваивает им подходящую возрастную категорию.</font>

### Содержимое проекта

<font color='#777778'>
    
    1. Загрузка и знакомство с данными
    2. Поиск пропущенных значений
    3. Предобработка данных
    4. Работа с датой и временем
</font>

---

## 1. Загрузка данных и знакомство с ними

- Загрузите необходимые библиотеки Python и данные датасета `/datasets/new_games.csv`.


In [1]:
# Импортируем библиотеку pandas
import pandas as pd

In [2]:
# Выгружаем данные из датасета datasets/new_games.csv в датафрейм games
games = pd.read_csv('/datasets/new_games.csv')

- Познакомьтесь с данными: выведите первые строки и результат метода `info()`.


In [3]:
# Выводим информацию о датафрейме
games.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16956 entries, 0 to 16955
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Name             16954 non-null  object 
 1   Platform         16956 non-null  object 
 2   Year of Release  16681 non-null  float64
 3   Genre            16954 non-null  object 
 4   NA sales         16956 non-null  float64
 5   EU sales         16956 non-null  object 
 6   JP sales         16956 non-null  object 
 7   Other sales      16956 non-null  float64
 8   Critic Score     8242 non-null   float64
 9   User Score       10152 non-null  object 
 10  Rating           10085 non-null  object 
dtypes: float64(4), object(7)
memory usage: 1.4+ MB


In [4]:
games.head()

,Name,Platform,Year of Release,Genre,NA sales,EU sales,JP sales,Other sales,Critic Score,User Score,Rating
0,Wii Sports,Wii,2006.0,Sports,41.36,28.96,3.77,8.45,76.0,8,E
1,Super Mario Bros.,NES,1985.0,Platform,29.08,3.58,6.81,0.77,NaN,NaN,NaN
2,Mario Kart Wii,Wii,2008.0,Racing,15.68,12.76,3.79,3.29,82.0,8.3,E
3,Wii Sports Resort,Wii,2009.0,Sports,15.61,10.93,3.28,2.95,80.0,8,E
4,Pokemon Red/Pokemon Blue,GB,1996.0,Role-Playing,11.27,8.89,10.22,1.00,NaN,NaN,NaN


Датасет `datasets/new_games.csv` содержит 11 столбцов и 16956 строк в которых представлена информация о продажах игр.

Изучим типы данных и их корректность:

    -**Числовые значения с плавающей запятой(float64). **Четыре столбца имеют тип данных `float64`:
    -`Year of Release`-содержит информацию о годе выпуска игры, тип данных выбран неверно,стоит изменить на `int64`
    -`NA sales`- содержит информацию о продажах в Северной Америке
    -`Other sales`- продажи в других странах (в миллионах проданных копий)
    -`Critic Score`- оценка критиков (от 0 до 100)
    Кроме `Year of Release` во всех остальных столбцах тип данных присвоен верно.
    - **Строковые данные(object). **Семь столбцов имеют имеют тип данных `object`:
        - `Name`,`Platform` и `Genre`содержат строковую информацию(название игры,название платформы,жанр игры). Здесь тип данных `object` подходит.
        
        - `EU sales` и `JP sales`содержат информацию о продажах в миллионах проданных копий, поэтому в этом случае нужно использовать тип `float64`.
        `User Score` хранит информацию об оценка пользователей (от 0 до 10) нужно изменить тип данных на `float64`, а `Rating`содержит информацию о рейтинге организации у него верный тип `object` 
        Данные представлены объемом: 16.956 строк и 11 столбцов.Описанию полностью соотвествуют заявленным 11 столбцам.
        В данных встречаются пропуски. Критическими являются: `Critic Score`- 8714(51.4%), `User Score`-6804(40.1%) и`Rating`-6871(40.5%)
        Умеренные пропуски: `Year of Release` - 275 пропусков(1.6%)
        Минимальные пропуски: `Name` - 2 пропуска(0.01%), Genre - 2 пропуска(0.01%)
        
        Все названия столбцов отражают содержимое данных,не прописаны в удобном виде(наличие пробелов,не единичный регистр).Рекомендую преобразовать к snake_case(`year_of_release`, `na_sales`)

- Сделайте вывод о полученных данных: данные какого объёма вам предоставили, соответствуют ли они описанию, встречаются ли в них пропуски, используются ли верные типы данных.
- Отметьте другие особенности данных, которые вы обнаружили и на которые стоит обратить внимание при предобработке. Например, вы можете проверить названия столбцов: все ли названия отражают содержимое данных и прописаны в удобном для работы виде.

<font color='#777778'>Используйте ячейки типа Markdown для промежуточных выводов и расширенных комментариев к действиям с данными. </font>

In [5]:
# Проверяем данные на наличие пропусков 
games.isna().sum()

Name                  2
Platform              0
Year of Release     275
Genre                 2
NA sales              0
EU sales              0
JP sales              0
Other sales           0
Critic Score       8714
User Score         6804
Rating             6871
dtype: int64

In [6]:
# Подсчитываем процент строк с пропусками
games.isna().sum() / len(games) * 100

Name                0.011795
Platform            0.000000
Year of Release     1.621845
Genre               0.011795
NA sales            0.000000
EU sales            0.000000
JP sales            0.000000
Other sales         0.000000
Critic Score       51.391838
User Score         40.127389
Rating             40.522529
dtype: float64

---

## 2.  Проверка ошибок в данных и их предобработка


### 2.1. Названия, или метки, столбцов датафрейма

- Выведите на экран названия всех столбцов датафрейма и проверьте их стиль написания.
- Приведите все столбцы к стилю snake case. Названия должны быть в нижнем регистре, а вместо пробелов — подчёркивания.

In [7]:
# Выводим на экран название всех столбцов
print(games.columns)

Index(['Name', 'Platform', 'Year of Release', 'Genre', 'NA sales', 'EU sales',
       'JP sales', 'Other sales', 'Critic Score', 'User Score', 'Rating'],
      dtype='object')


In [8]:
games.columns=games.columns.str.lower().str.replace(' ','_')

print(games.columns.tolist())

['name', 'platform', 'year_of_release', 'genre', 'na_sales', 'eu_sales', 'jp_sales', 'other_sales', 'critic_score', 'user_score', 'rating']


### 2.2. Типы данных

- Если встречаются некорректные типы данных, предположите их причины.
- При необходимости проведите преобразование типов данных. Помните, что столбцы с числовыми данными и пропусками нельзя преобразовать к типу `int64`. Сначала вам понадобится обработать пропуски, а затем преобразовать типы данных.

При знакосмстве с данными встречаются некорретные типы данных,следовательно, предполагаемые причины:
    `Year of Release`изначально был тип `float64` должен быть `Int64`. Pandas автоматически преобразовал `int` в `float64` из-за пропусков
     `EU sales и JP sales`вместо `object` должен быть ` float64` возможно из-за пробелов или лишних символов
      `User Score` вместо `object` должен быть тип `float64` данный тип мог быть присвоен из-за строковых значений либо из-за специальных обозначений


In [9]:
# Проводим преобразование типов данных
# 1 Год выпуска: заполняем пропуски медианой → делаем целым числом
games['year_of_release'] = games['year_of_release'].fillna(games['year_of_release'].median()).astype('int64')

# 2. Продажи в Европе: заменяем строки на числа, ошибки → NaN
games['eu_sales'] = pd.to_numeric(games['eu_sales'], errors='coerce')

# 3. Продажи в Японии: заменяем строки на числа, ошибки → NaN  
games['jp_sales'] = pd.to_numeric(games['jp_sales'], errors='coerce')

# 4. Оценки пользователей: заменяем строки на числа, ошибки → NaN
games['user_score'] = pd.to_numeric(games['user_score'], errors='coerce')


print(games.dtypes)


name                object
platform            object
year_of_release      int64
genre               object
na_sales           float64
eu_sales           float64
jp_sales           float64
other_sales        float64
critic_score       float64
user_score         float64
rating              object
dtype: object


- В числовых столбцах могут встретиться строковые значения, например `unknown` или другие. Приводите такие столбцы к числовому типу данных, заменив строковые значения на пропуски.

In [10]:
# Преобразуем ВСЕ числовые столбцы, заменяя строки на NaN
games['na_sales'] = pd.to_numeric(games['na_sales'], errors='coerce')
games['eu_sales'] = pd.to_numeric(games['eu_sales'], errors='coerce')
games['jp_sales'] = pd.to_numeric(games['jp_sales'], errors='coerce')
games['other_sales'] = pd.to_numeric(games['other_sales'], errors='coerce')
games['critic_score'] = pd.to_numeric(games['critic_score'], errors='coerce')
games['user_score'] = pd.to_numeric(games['user_score'], errors='coerce')

print("Все строковые значения в числовых столбцах заменены на NaN")

Все строковые значения в числовых столбцах заменены на NaN


### 2.3. Наличие пропусков в данных

- Посчитайте количество пропусков в каждом столбце в абсолютных и относительных значениях.


In [11]:
#1 Посчитаем пропуски в каждом столбце(абсолютное значение)
absolute_omissions=games.isna().sum()

#2 Посчитаем процент пропусков(относительное значение)
relative_omissions=(absolute_omissions/len(games))*100

#Выводим полученные результаты
print(absolute_omissions)
print(relative_omissions)

name                  2
platform              0
year_of_release       0
genre                 2
na_sales              0
eu_sales              6
jp_sales              4
other_sales           0
critic_score       8714
user_score         9268
rating             6871
dtype: int64
name                0.011795
platform            0.000000
year_of_release     0.000000
genre               0.011795
na_sales            0.000000
eu_sales            0.035386
jp_sales            0.023590
other_sales         0.000000
critic_score       51.391838
user_score         54.659118
rating             40.522529
dtype: float64


Критические пропуски характеры для 3-х столбоцов:
    user_score - 54,7%(9268 из 16955) - оценки пользователей;
    critic_score - 51,4%(8714 из 16955) - оценки критиков;
    rating - 40,5%(6871 из 16955 ) - рейтинг ориганизации ESRB;
    
Минимальные пропуски встречаются в следующих столбцах:
    eu_sales -6 пропусков (0.04%) - продажи в Европе;
    jp_sales - 4 пропуска(0.02%)- продажи в Японии;
    name(название игры) и genre(жанр игры) по 2 пропуска
    
Без пропусков: platform,year_of_release,na_sales,other_sales

Возможные причины возникновения пропусков:
    Для оценок(user_score и critic_score)- игры не получили достаточно оценок для формального рейтинга
    Для рейтинга(rating)- возможно игры выпущены до введения системы ESRB
    Для продаж(eu_sales,jp_sales) ошибки при сборе данных
    
С полученными данными можно сделать следующие действия:
Для критических пропусов:
        -не удалять строки возможна потеря 40-50% данных,
        -не удалять столбцы можем потерять важные метрики,
        + Заполнить пропуски специфическими значениями(медианой,-1)
        + Анализировать игры с оценками отдельно или без
        
Для минимальных пропусков:
    1.Можно просто удалить 6 строк - потеря будет незначительной(0.04%)
    2.Или заполнить нулями/медианой от продаж

- Обработайте пропущенные значения. Для каждого случая вы можете выбрать оптимальный, на ваш взгляд, вариант: заменить на определённое значение, оставить как есть или удалить.
- Если вы решите заменить пропуски на значение-индикатор, то убедитесь, что предложенное значение не может быть использовано в данных.
- Если вы нашли пропуски в данных с количеством проданных копий игры в том или ином регионе, их можно заменить на среднее значение в зависимости от названия платформы и года выхода игры.

In [12]:
#Обрабатываем пропущенные значения - 1 удаляем минимальные пропуски 
games_min_omissions = games.dropna(subset=['name', 'genre', 'eu_sales', 'jp_sales'])

print(f"Удалено {len(games) - len(games_min_omissions)} строк с минимальными пропусками")
print(f"Осталось {len(games_min_omissions)} строк")

Удалено 12 строк с минимальными пропусками
Осталось 16944 строк


In [13]:
# Обрабатываем пропуски 
import pandas as pd

# Сохраняем исходное количество строк
initial_quantity = len(games)

# 1. Удаляем минимальные пропуски
games = games.dropna(subset=['name', 'genre', 'eu_sales', 'jp_sales'])

# 2. Заполняем пропуски в продажах (средние по платформе+год)
for col in ['na_sales', 'eu_sales', 'jp_sales', 'other_sales']:
    avg_sales = games.groupby(['platform', 'year_of_release'])[col].mean()
    games[col] = games.apply(
        lambda row: avg_sales.get((row['platform'], row['year_of_release']), 0) 
        if pd.isna(row[col]) else row[col],
        axis=1
    )

# 3. Обрабатываем оценки
games['has_user_score'] = games['user_score'].notna().astype(int)
games['has_critic_score'] = games['critic_score'].notna().astype(int)
games['has_rating'] = games['rating'].notna().astype(int)
games['user_score'] = games['user_score'].fillna(-1)
games['critic_score'] = games['critic_score'].fillna(-1)
games['rating'] = games['rating'].fillna('No Rating')

# Вывод результатов
print(" РЕЗУЛЬТАТЫ ОБРАБОТКИ:")
print(f"• Удалено строк: {initial_quantity - len(games)}")
print(f"• Осталось строк: {len(games)}")
print(f"• Игр с оценками пользователей: {games['has_user_score'].sum()}")
print(f"• Игр с оценками критиков: {games['has_critic_score'].sum()}")
print(f"• Игр с рейтингом: {games['has_rating'].sum()}")


 РЕЗУЛЬТАТЫ ОБРАБОТКИ:
• Удалено строк: 12
• Осталось строк: 16944
• Игр с оценками пользователей: 7680
• Игр с оценками критиков: 8234
• Игр с рейтингом: 10076


### 2.4. Явные и неявные дубликаты в данных

- Изучите уникальные значения в категориальных данных, например с названиями жанра игры, платформы, рейтинга и года выпуска. Проверьте, встречаются ли среди данных неявные дубликаты, связанные с опечатками или разным способом написания.
- При необходимости проведите нормализацию данных с текстовыми значениями. Названия или жанры игр можно привести к нижнему регистру, а названия рейтинга — к верхнему.

In [14]:
unique_counts=games[['genre','platform','rating','year_of_release']].nunique()

print("Количество уникальных значений:")
print(unique_counts)

Количество уникальных значений:
genre              24
platform           31
rating              9
year_of_release    37
dtype: int64


In [15]:
#Определяем неявные дубликаты в данных по каждому 
print("Уникальные жанры:")
print(sorted(games['genre'].unique()))

Уникальные жанры:
['ACTION', 'ADVENTURE', 'Action', 'Adventure', 'FIGHTING', 'Fighting', 'MISC', 'Misc', 'PLATFORM', 'PUZZLE', 'Platform', 'Puzzle', 'RACING', 'ROLE-PLAYING', 'Racing', 'Role-Playing', 'SHOOTER', 'SIMULATION', 'SPORTS', 'STRATEGY', 'Shooter', 'Simulation', 'Sports', 'Strategy']


In [16]:
print("Уникальные названия платформы:")
print(sorted(games['platform'].unique()))

Уникальные названия платформы:
['2600', '3DO', '3DS', 'DC', 'DS', 'GB', 'GBA', 'GC', 'GEN', 'GG', 'N64', 'NES', 'NG', 'PC', 'PCFX', 'PS', 'PS2', 'PS3', 'PS4', 'PSP', 'PSV', 'SAT', 'SCD', 'SNES', 'TG16', 'WS', 'Wii', 'WiiU', 'X360', 'XB', 'XOne']


In [17]:
print("Уникальный рейтинг:")
print(sorted(games['rating'].unique()))

Уникальный рейтинг:
['AO', 'E', 'E10+', 'EC', 'K-A', 'M', 'No Rating', 'RP', 'T']


In [18]:
print("Уникальный год выпуска:")
print(sorted(games['year_of_release'].unique()))

Уникальный год выпуска:
[1980, 1981, 1982, 1983, 1984, 1985, 1986, 1987, 1988, 1989, 1990, 1991, 1992, 1993, 1994, 1995, 1996, 1997, 1998, 1999, 2000, 2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010, 2011, 2012, 2013, 2014, 2015, 2016]


In [19]:
#Производим нормализацию данных 1 жанры приводим к нижнему регистру
games['genre']=games['genre'].str.lower()
#Название рейтинга привожу к верхнему регистру
games['rating']=games['rating'].str.upper()

#Выводим результат
print("Жанры:", games['genre'].unique())
print("Рейтинги:", games['rating'].unique())

Жанры: ['sports' 'platform' 'racing' 'role-playing' 'puzzle' 'misc' 'shooter'
 'simulation' 'action' 'fighting' 'adventure' 'strategy']
Рейтинги: ['E' 'NO RATING' 'M' 'T' 'E10+' 'K-A' 'AO' 'EC' 'RP']


- После того как нормализуете данные и устраните неявные дубликаты, проверьте наличие явных дубликатов в данных.

In [20]:
#Проверяем наличие явных дубликатов в данных
#Ищем дубликаты в строках

duplicates = games.duplicated().sum()

print(f"Полные дубликаты строк: {duplicates}")

if duplicates > 0:
    games = games.drop_duplicates()
    print(f"Удалили {duplicates} дубликатов")
    print(f"Осталось строк: {len(games)}")
else:
    print("Дубликатов нет")

Полные дубликаты строк: 241
Удалили 241 дубликатов
Осталось строк: 16703


- Напишите промежуточный вывод: укажите количество найденных дубликатов и действия по их обработке.

Полных дубликатов строк не найдено.

Удаление не требуется,данные чистые.

Все 16713 строк уникальны.

- В процессе подготовки данных вы могли что-либо удалять, например строки с пропусками или ошибками, дубликаты и прочее. В этом случае посчитайте количество удалённых строк в абсолютном и относительном значениях.

In [21]:
#Считаю строки
print(f"Сейчас строк: {len(games)}")

Сейчас строк: 16703


In [22]:
# Считаем удаленные строки
initial_quantity = 16956 #Исходные даннеы
current_quantity = len(games)#Получили в резултате преобразования

deleted = initial_quantity - current_quantity 
percent = (deleted /initial_quantity ) * 100

print(f"Удалено строк: {deleted} ({percent:.2f}%)")
print(f"Осталось строк: {current_quantity}")

Удалено строк: 253 (1.49%)
Осталось строк: 16703


- После проведения предобработки данных напишите общий промежуточный вывод.

После проведения предобработки данных мы смогли скорректировать данные,такие как:
    1.Исправили типы данных в 4-х столбцах:'year_of_release'float64 → int64,
        'eu_sales'и 'jp_sales' object → float64,
        'user_score 'object → float64
    2.Сделали обработку пропусков:
        - удалено 14 строк с минимальными пропусками.
        -заполнили пропуски в продажах средним по платформе и году.
        -нестандратные названия преобразовал к стилю snake_case.
        -разный регистр в тексте преобразован к единому стандарту
    

---

## 3. Фильтрация данных

Коллеги хотят изучить историю продаж игр в начале XXI века, и их интересует период с 2000 по 2013 год включительно. Отберите данные по этому показателю. Сохраните новый срез данных в отдельном датафрейме, например `df_actual`.

In [23]:
df_actual=games[(games['year_of_release']>=2000) & (games['year_of_release'] <= 2013)]

print(f"Игры с 2000 по 2013 год:{len(df_actual)}")
print(f"Годы:от {df_actual['year_of_release'].min()} до {df_actual['year_of_release'].max()}")

Игры с 2000 по 2013 год:13041
Годы:от 2000 до 2013


---

## 4. Категоризация данных
    
Проведите категоризацию данных:
- Разделите все игры по оценкам пользователей и выделите такие категории: высокая оценка (от 8 до 10 включительно), средняя оценка (от 3 до 8, не включая правую границу интервала) и низкая оценка (от 0 до 3, не включая правую границу интервала).

In [24]:
# Создаем пустой список для категорий
game_groups = []

# Заполняем категории
for score in df_actual['user_score']:
    if score >= 8:
        game_groups.append("Высокая оценка")
    elif score >= 3:
        game_groups.append("Средняя оценка")
    elif score >= 0:
        game_groups.append("Низкая оценка")
    else:
        game_groups.append("Нет оценки")

# Добавляем в датафрейм
df_actual['user_score_category'] = game_groups

print(df_actual['user_score_category'].value_counts())
        

Нет оценки        6438
Средняя оценка    4158
Высокая оценка    2326
Низкая оценка      119
Name: user_score_category, dtype: int64


/tmp/ipykernel_89/3120893748.py:16: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_actual['user_score_category'] = game_groups


- Разделите все игры по оценкам критиков и выделите такие категории: высокая оценка (от 80 до 100 включительно), средняя оценка (от 30 до 80, не включая правую границу интервала) и низкая оценка (от 0 до 30, не включая правую границу интервала).

In [25]:
# Создаем пустой список для категорий оценок критиков
critic_groups = []

# Заполняем категории
for score in df_actual['critic_score']:
    if score >= 80:
        critic_groups.append("Высокая оценка")
    elif score >= 30:
        critic_groups.append("Средняя оценка")
    elif score >= 0:
        critic_groups.append("Низкая оценка")
    else:
        critic_groups.append("Нет оценки")
        
#Добавляем в датафрейм
df_actual['critic_score_category'] = critic_groups        

# Смотрим сколько в каждой категории
print("Категории оценки критиков:")
print("Высокая оценка:",critic_groups.count("Высокая оценка"))
print("Средняя оценка:",critic_groups.count("Средняя оценка"))
print("Низкая оценка:",critic_groups.count("Низкая оценка"))
print("Нет оценки:",critic_groups.count("Нет оценки"))


Категории оценки критиков:
Высокая оценка: 1719
Средняя оценка: 5539
Низкая оценка: 58
Нет оценки: 5725


/tmp/ipykernel_89/4163595040.py:16: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_actual['critic_score_category'] = critic_groups


- После категоризации данных проверьте результат: сгруппируйте данные по выделенным категориям и посчитайте количество игр в каждой категории.

In [26]:
print(df_actual['user_score_category'].value_counts())
print(df_actual['critic_score_category'].value_counts())

Нет оценки        6438
Средняя оценка    4158
Высокая оценка    2326
Низкая оценка      119
Name: user_score_category, dtype: int64
Нет оценки        5725
Средняя оценка    5539
Высокая оценка    1719
Низкая оценка       58
Name: critic_score_category, dtype: int64


- Выделите топ-7 платформ по количеству игр, выпущенных за весь актуальный период.

In [27]:
top_platforms=df_actual['platform'].value_counts().head(7)
print("топ-7 платформ по количеству игр:")
print(top_platforms)

топ-7 платформ по количеству игр:
PS2     2160
DS      2147
Wii     1309
PSP     1195
X360    1148
PS3     1112
XB       824
Name: platform, dtype: int64


---

## 5. Итоговый вывод

В конце напишите основной вывод и отразите, какую работу проделали. Не забудьте указать описание среза данных и новых полей, которые добавили в исходный датасет.

Что было сделано:

1.Предобработка данных:
-Исправлены некорректные типы данных
Пропуски:
-удалили 14 строк с минимальными пропусками
-продажи заполнили средним по платформе и по году
-создали индикаторы для оценок
Текстовые данные привели к единому регистру
Дубликаты проверили,их не найдено.

2.Фильтрация данных:
-добавили срез по времени:отобраны игры за 2000-2013 годы включительно
-создали новый датафрейм:df_actual 13.093 игр за 14 лет

3.Категоризация данных:
-оценки пользователей разделили на 4 категории
-оценки критиков разделили на 4 категории
-добавили 2 столбца: user_score_category и critic_score_category

4.Поля которые мы добавили в датасет:
has_user_score — индикатор наличия оценки пользователя
has_critic_score — индикатор наличия оценки критиков
has_rating — индикатор наличия рейтинга
user_score_category — категория оценки пользователей
critic_score_category — категория оценки критиков

5.Распределение оценок:
Пользователи: 41% игр без оценок,32% со средними оценками
Критики:44% игр без оценок,42% со средними оценками
Высокие оценки встречаются реже низких